app/src/main/java/com/example/demo_music_player/ui/MainActivity.kt

```diff
@@ -0,0 +1,145 @@
+        }
+
+        override fun onServiceDisconnected(className: ComponentName) {
+            playerPresenter.detach()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        setOnclickButtons()
+        prepareStartUIForDemo()
+        startService()
+        playerPresenter = PlayerPresenter(this, Songs())
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        stopService()
+    }
+
+    override fun setProgressBar(currentTime: Int) {
+        textCurrentTime?.text = TimeUtils.millisecondsToString(currentTime)
+        seekSongProgress?.progress = currentTime
+    }
+
+    override fun setPresenter(presenter: PlayerContract.Presenter) {
+        playerPresenter = presenter
+    }
+
+    override fun setStatus(status: Boolean) {
+        this.playerStatus = status
+    }
+
+    override fun setCurrentSongIndex(index: Int) {
+        this.currentSongIndex = index
+    }
+
+    override fun setCurrentSong(newSong: String) {
+        this.currentSongName = newSong
+    }
+
+    override fun getStatus() = this.playerStatus
+
+    override fun getCurrentSongIndex() = this.currentSongIndex
+
+    override fun getCurrentSong()= this.currentSongName
+
+    override fun setButtonBackground(status: Boolean) {
+        val newButtonPlayResource = if (status) R.drawable.player_play_button else R.drawable.player_pause_button
+        buttonPlay?.setBackgroundResource(newButtonPlayResource)
+    }
+
+    override fun setUI() {
+        textSongTitle?.text = currentSongName
+        textCurrentTime?.text = EMPTY_TIME
+        seekSongProgress?.progress = 0
+        seekSongProgress?.max = playerPresenter.getSongDuration()
+        textDuration?.text = TimeUtils.millisecondsToString(playerPresenter.getSongDuration())
+    }
+
+    override fun getRawResIdByName(name: String): Int {
+        val packageName: String = this.packageName
+        return resources.getIdentifier(name, "raw", packageName)
+    }
+
+    private fun prepareStartUIForDemo() {
+        currentSongName = START_SONG_NAME
+        textDuration?.text = START_SONG_DURATION_TIME
+        textCurrentTime?.text = EMPTY_TIME
+        seekSongProgress?.max = START_SONG_DURATION
+        textSongTitle?.text = currentSongName
+    }
+
+    private fun setOnclickButtons() {
+        buttonPlay.setOnClickListener {
+            playerPresenter.playButtonClicked()
+        }
+        buttonNext.setOnClickListener {
+            playerPresenter.nextButtonClicked()
+        }
+        buttonPrevious.setOnClickListener {
+            playerPresenter.previousButtonClicked()
+        }
+    }
+
+    private fun doBindService() {
+        bindService(
+            Intent(this, ForegroundService::class.java),
+            mConnection, Context.BIND_AUTO_CREATE
+        )
+    }
+
+    private fun doUnbindService() {
```

Comment:
do chỉ đi kèm với homework, housework, exercise
```suggestion
    private fun unbindService()  = unBindService(serviceConnection)
```
tương tự với bindService()